experiments/AMIP/coupler_driver.jl

```diff
@@ -60,7 +60,7 @@ using ClimaCoupler.FluxCalculator:
     MoninObukhovScheme,
     partitioned_turbulent_fluxes!,
     water_albedo_from_atmosphere!
-using ClimaCoupler.Interfacer: CoupledSimulation, SurfaceStub, get_field, update_field!
+using ClimaCoupler.Interfacer: CoupledSimulation, SurfaceStub, get_field, update_field!, step!
```

Comment:
Do we need `step!` here?

Reply:
Yes, otherwise when we call `step!` on `ocean_sim` or `ice_sim` when they're `SurfaceStub`s, we don't have access to the `SurfaceStub` method of `step!`

Reply:
Ahh, I see. It was defined locally in the FieldExchanger previously. Could we make `step!` and `reinit!` functions of the`Interfacer`, which throw an error if not defined? Then then we don't have to expose the `step!` function in the driver, where it is not directly used.

Reply:
It looks like `step!` is used directly in the driver, during our initial component model exchange ([here](https://github.com/CliMA/ClimaCoupler.jl/blob/main/experiments/AMIP/coupler_driver.jl#L625-L627)). `step!` and `reinit!` are functions of the Interfacer even though they're in `surface_stub.jl`, since that file is included by `Interfacer.jl`. I think we didn't explicitly import these functions in the driver before because they were already imported by files we include by the driver (e.g. atmos_init), but we use `step!` in the driver too.

I can add fallback methods to the Interfacer that error if they aren't extended

Reply:
I meant `step!` of the stub isn't directly used. But yeah, it would be nice for all of the `step!` functions to be extensions of the Interfacer fallbacks. :) I guess we were shortcutting this before with the `step!` of the stub being defined inside the FieldExchanger module, where all the `step!`s are called within the src code. 👀

Reply:
I think `step!` of a stub is used in the driver, in line 626 that I linked, since `ocean_sim` is a `SurfaceStub` in AMIP sims. Maybe I'm missing something?

Reply:
Nope, that was me. You're right, it is used. :)

Reply:
Okay! Good call with adding the fallback methods for these functions :) Are you okay with this PR as it is now?

Reply:
Yeah, looks great. After the import fix, feel free to merge. Thanks, Julia!